src/store/operators/makeThunkAsyncRequest.ts

```diff
@@ -0,0 +1,19 @@
+import { ThunkDispatch } from 'redux-thunk';
+import { Action } from 'redux';
+import { RootState } from '../types';
+import { failureAction, successAction } from '../actionTypes';
+
+const makeThunkAsyncRequest =
+  (action: Function, actionType: string) => (data?: any) => (dispatch: ThunkDispatch<RootState, void, Action>) => {
```

Comment:
Suggestion: I think this declaration of the `data` parameter is only a single parameter. I would use something like `(...data:any[])`, but it would be even better if you will be able to transform it to the template type.

Reply:
Also looks like the `action` is not a simple function, but a function returning a promise, right?